source/pkg/reconciler/rabbitmqsource.go

```diff
@@ -0,0 +1,241 @@
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes"
+	appsv1listers "k8s.io/client-go/listers/apps/v1"
+	"knative.dev/eventing-rabbitmq/source/pkg/client/clientset/versioned"
+	"knative.dev/eventing-rabbitmq/source/pkg/apis/sources/v1alpha1"
+	reconcilerrabbitmqsource "knative.dev/eventing-rabbitmq/source/pkg/client/injection/reconciler/sources/v1alpha1/rabbitmqsource"
+	listers "knative.dev/eventing-rabbitmq/source/pkg/client/listers/sources/v1alpha1"
+	"knative.dev/eventing-rabbitmq/source/pkg/reconciler/resources"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/logging"
+	pkgLogging "knative.dev/pkg/logging"
+	"knative.dev/pkg/metrics"
+	pkgreconciler "knative.dev/pkg/reconciler"
+	"knative.dev/pkg/resolver"
+)
+
+const (
+	raImageEnvVar              		= "RABBITMQ_RA_IMAGE"
+	rabbitmqReadinessChanged   		= "RabbitmqSourceReadinessChanged"
+	rabbitmqUpdateStatusFailed      = "RabbitmqSourceUpdateStatusFailed"
+	rabbitmqSourceDeploymentCreated = "RabbitmqSourceDeploymentCreated"
+	rabbitmqSourceDeploymentUpdated = "RabbitmqSourceDeploymentUpdated"
+	rabbitmqSourceDeploymentFailed  = "RabbitmqSourceDeploymentUpdated"
+	rabbitmqSourceReconciled        = "RabbitmqSourceReconciled"
+	component       				= "rabbitmqsource"
+)
+
+var (
+	deploymentGVK = appsv1.SchemeGroupVersion.WithKind("Deployment")
+)
+
+func newDeploymentCreated(namespace, name string) pkgreconciler.Event {
+	return pkgreconciler.NewEvent(corev1.EventTypeNormal, rabbitmqSourceDeploymentCreated, "RabbitmqSource created deployment: \"%s/%s\"", namespace, name)
+}
+
+func deploymentUpdated(namespace, name string) pkgreconciler.Event {
+	return pkgreconciler.NewEvent(corev1.EventTypeNormal, rabbitmqSourceDeploymentUpdated, "RabbitmqSource updated deployment: \"%s/%s\"", namespace, name)
+}
+
+func newDeploymentFailed(namespace, name string, err error) pkgreconciler.Event {
+	return pkgreconciler.NewEvent(corev1.EventTypeWarning, rabbitmqSourceDeploymentFailed, "RabbitmqSource failed to create deployment: \"%s/%s\", %w", namespace, name, err)
+}
+
+type Reconciler struct {
+	// KubeClientSet allows us to talk to the k8s for core APIs
+	KubeClientSet kubernetes.Interface
+
+	receiveAdapterImage string
+
+	rabbitmqLister	 listers.RabbitmqSourceLister
+	deploymentLister appsv1listers.DeploymentLister
+
+	rabbitmqClientSet   versioned.Interface
+	loggingContext		context.Context
+	loggingConfig 		*pkgLogging.Config
+	metricsConfig 		*metrics.ExporterOptions
+
+	sinkResolver		*resolver.URIResolver
+}
+
+var _ reconcilerrabbitmqsource.Interface  = (*Reconciler)(nil)
+
+func (r *Reconciler) ReconcileKind(ctx context.Context, src *v1alpha1.RabbitmqSource) pkgreconciler.Event {
```

Comment:
Don't you also need to have FinalizeKind for any cleanup? Just jotting this down so I don't forget :)